build.gradle.kts

```
@@ -95,20 +95,17 @@ spotless {
 }
 
 tasks.withType<JavaCompile>().configureEach {
-    options.errorprone {
-        disableWarningsInGeneratedCode.set(true)
-        option("NullAway:AnnotatedPackages", "com.mongodb.hibernate")
+    if (name == "compileJava") {
+        options.errorprone {
+            disableWarningsInGeneratedCode.set(true)
+            option("NullAway:AnnotatedPackages", "com.mongodb.hibernate")
+            error("NullAway")
+        }
+        options.compilerArgs.addAll(listOf("-Xlint:deprecation", "-Werror"))
+    } else {
+        options.errorprone.isEnabled.set(false)
     }
```

Let's do `options.compilerArgs.addAll(listOf("-Xlint:deprecation", "-Werror"))` for all compile tasks, not only for production code

```suggestion
    options.compilerArgs.addAll(listOf("-Xlint:deprecation", "-Werror"))
    if (name == "compileJava") {
        options.errorprone {
            disableWarningsInGeneratedCode.set(true)
            option("NullAway:AnnotatedPackages", "com.mongodb.hibernate")
            error("NullAway")
        }
    } else {
        options.errorprone.isEnabled.set(false)
    }
```